File: readability/readability.c
//https://cs50.harvard.edu/x/2022/psets/2/readability/
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    double lettersAmount = count_letters(text);
    double wordsAmount = count_words(text);
    double sentencesAmount = count_sentences(text);
    double index = (0.0588 * ((lettersAmount/wordsAmount)*100)) - (0.296 * ((sentencesAmount/wordsAmount)*100)) - 15.8;
    int output = round(index);
    index>=(double)16? printf("Grade 16+") : index<(double)1 ? printf("Before Grade 1") : printf("Grade %d", output);
    printf("\n");
}

int count_letters(string text)
{
    int lettersAmount = 0;
    for(int i = 0; i<=strlen(text+1); i++){
        if((text[i]>='A'&&text[i]<='Z')||(text[i]>='a'&&text[i]<='z')) lettersAmount++;
    }
    return lettersAmount;
}

int count_words(string text)
{
    int wordsAmount = 1;
    for(int i = 0; i<=strlen(text+1); i++){
        if(text[i]==' ') wordsAmount++;
    }
    return wordsAmount;
}

int count_sentences(string text)
{
    int sentencesAmount = 0;
    for(int i = 0; i<=strlen(text+1); i++){
        if(text[i]=='.'||text[i]=='!'||text[i]=='?') sentencesAmount++;

    }
    return sentencesAmount;
}

File: sentimental-readability/readability.py
from cs50 import get_string

def formula(letter_count, word_count, sentence_count):
    l = 100 * letter_count / word_count
    s = 100 * sentence_count / word_count
    result = (0.0588 * l) - (0.296 * s) - 15.8
    return round(result)

text = get_string("Text: ")

count = 0
letter_count = 0
word_count = 0
sentence_count = 0

for c in text:
    if c.isalpha():
        letter_count += 1
    elif c==' ':
        word_count += 1
    elif c=='.' or c=='!' or c=='?':
        sentence_count += 1

word_count += 1

index = formula(letter_count, word_count, sentence_count)

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(("Grade {}").format(index))